core/app/models/concerns/spree/user_methods.rb

```diff
@@ -34,6 +36,14 @@ def has_spree_role?(role_in_question)
       spree_roles.any? { |role| role.name == role_in_question.to_s }
     end
 
+    def generate_api_key(role: nil)
```

Comment:
I think the naming here is a little ambiguous, I don't think a developer would be able to infer that `user.generate_spree_api_key` always generates an api key, but `user.generate_api_key` only generates one if the user is eligible. Maybe a name like `auto_generate_spree_api_key` or similar would clear that up.

I'd also love if this looked through the roles rather than needing the role to be passed in, just to keep the api as simple as possible. We have the knowledge in this object of what roles we belong to so we might as well use that. Maybe something like `(spree_roles.map(&:name) & Spree::Config.roles_for_auto_api_key).any?`.

Other than those two comments this is awesome :+1: